### DISCUSSION TITLE
Compiling stops with error on Manjaro Linux

### DISCUSSION THREAD

**Discussion by hkalle** [awawa-dev/HyperHDR#257] [Q&A]
My HTPC is runing on the latest Manjaro Linux 21.2.6 and I was succesfully compiling last time on 15 Dec 2021 following the guidelines provided for Arch Linux.

Following now the same procedure the build fails with an error and according to my interpretation it seems to be related to the qt5/qt6 libaries as pointed out in the make.log-file attached:

[cmake.log](https://github.com/awawa-dev/HyperHDR/files/8530818/cmake.log)
[make.log](https://github.com/awawa-dev/HyperHDR/files/8530819/make.log)

Anyone who could point me to the rigth direction? Thanks!

**Comment by awawa-dev** [comment]
Interesting. 
> error "Qt requires a C++17 compiler"  

This QT6 is missing some important plugins like serial port, so maybe something else is configured in specific way. Previously you use QT6 or QT5? Could you downgrade to QT5? If you already have QT5 dev libs installed you can provide option DO_NOT_USE_QT_VERSION_6_LIBS for HyperHDR's cmake configuration.  ENABLE_PIPEWIRE can be also set to OFF just for checking if it helps.

QT6 is used for WIndows and macOS builds but who knows?  But probably something is missing from the build system.

**Comment by hkalle** [comment]
Awawa, thanks for your quick support. Very much appriciated!

Option DO_NOT_USE_QT_VERSION_6_LIBS and ENABLE_PIPEWIRE did not solve the isssue. Compiler was still trying to link QT6 libraries. I could not downgrade to QT5, due to dependencies to Kvantum, but installed a basic Arch Linux system in a Virtual Box and it worked-out pretty well.

The only issue to point-out is, that python3 is a dependency to be installed. Maybe you would like to change it in your documentation.

**Comment by awawa-dev** [comment]
DO_NOT_USE_QT_VERSION_6_LIBS will work only if there are both QT5 and QT6 dev libs installed in the system. So I guess only QT6 was installed and it should be in the output of cmake configuration.
Could you provide more details about the problem with missing python3 dependencies? HyperHDR doesn't use python at all but maybe some dependent libraries try to use it.

**Comment by hkalle** [comment]
Please find the log were the build stops:

[  0%] Automatic MOC for target flatc
[  0%] Automatic MOC for target flathash
[  1%] Automatic MOC for target flatbuffers
[  1%] Automatic MOC for target mbedcrypto
[  1%] Built target flatc_autogen
[  1%] Built target flathash_autogen
[  1%] Built target flatbuffers_autogen
[  1%] Built target mbedcrypto_autogen
[  2%] Automatic MOC for target bonjour
[  2%] Automatic MOC for target effectengine
[  2%] Automatic MOC for target blackborder
[  3%] Automatic MOC for target ssdp
[  3%] Built target blackborder_autogen
[  3%] Built target bonjour_autogen
[  3%] Built target ssdp_autogen
[  3%] Automatic MOC for target boblightserver
[  3%] Automatic MOC for target leddevice
[  3%] Automatic MOC for target hyperhdr-utils
[  3%] Built target boblightserver_autogen
[  3%] Built target leddevice_autogen
[  3%] Built target effectengine_autogen
[  3%] Built target hyperhdr-utils_autogen
[  5%] Automatic MOC for target database
[  5%] Automatic MOC for target webserver
[  5%] Automatic MOC for target hyperhdr-api
[  5%] Automatic MOC for target smartPipewire
[  5%] Built target hyperhdr-api_autogen
[  5%] Built target webserver_autogen
[  5%] Built target database_autogen
[  5%] Built target smartPipewire_autogen
[  5%] Automatic MOC for target smartX11
[  5%] Automatic MOC for target resources
[  5%] Built target smartX11_autogen
[  5%] Built target resources_autogen
[  6%] Built target flatbuffers
[ 13%] Built target flatc
[ 13%] Generating ssl_debug_helpers_generated.h, ssl_debug_helpers_generated.c
[ 14%] Built target flathash
/usr/bin/env: ‘python3’: No such file or directory
make[2]: *** [dependencies/external/mbedtls/library/CMakeFiles/mbedcrypto.dir/build.make:207: dependencies/external/mbedtls/library/ssl_debug_helpers_generated.h] Error 127
make[1]: *** [CMakeFiles/Makefile2:841: dependencies/external/mbedtls/library/CMakeFiles/mbedcrypto.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
Consolidate compiler generated dependencies of target smartPipewire
[ 15%] Building CXX object sources/grabber/X11/CMakeFiles/smartX11.dir/smartX11_autogen/mocs_compilation.cpp.o
[ 15%] Building CXX object sources/grabber/X11/CMakeFiles/smartX11.dir/smartX11.cpp.o
[ 16%] Built target smartPipewire
[ 16%] Linking CXX shared library ../../../lib/libsmartX11.so
[ 16%] Built target smartX11
make: *** [Makefile:156: all] Error 2

**Comment by awawa-dev** [comment]
As in the log: the problem is in the external MBEDTLS library despite it didn't complain about any missing dependencies during its configuration phase which should stop the process then. At least I suspect it because I don't see these logs. I can extend the HyperHDR docs but it's a bad behavior from that library.
